### ISSUE TITLE
Keep aspect ratio modifier

### ISSUE THREAD

**Issue by RealyUniqueName** [jarnik/transformtool#2]
Hi! Great tool )
One feature i miss is ability to keep aspect ratio while transforming. E.g. in Inkscape you can press Ctrl to resize symbol without distorting it.

**Comment by impaler** [comment]
Yes this tool is great to see in haxe, I used the as3 version a couple of years ago and it had this ratio lock feature.

**Comment by jarnik** [comment]
Unfortunately, I've used a bit out-dated version of the AS3 TransformTool which came without the ability to maintain aspect.

It's a handy feature, it's definitely on my TODO list. 
I'm thinking of making the #1 scaling work similarly - hold SHIFT to scale symmetrically around the registration point.

**Comment by impaler** [comment]
Cool, so are you are planning to port version 2 instead? I imagine it may be easier in the long run, it had other features like scale size limits etc from what I remember.

Yes Shift is a pretty universal shortcut among programs for this.

**Comment by jarnik** [comment]
You are right, it would be more reasonable to work on the latest version.

I'll start over again from this one http://www.senocular.com/flash/tutorials/transformtool/

**Comment by impaler** [comment]
I appreciate you sharing your work. I am curious were there complications in porting the transform matrix apis with nme? Cheers.

**Comment by jarnik** [comment]
The only issues I've ran into with the out-dated version were the absence of a universal Dictionary class (I wrote a custom Hash<DisplayObject> table) and a few mouse-API-specific glitches.
